src/local_scheduler/local_scheduler.cc

```diff
@@ -78,28 +77,25 @@ int force_kill_worker(event_loop *loop, timer_id id, void *context) {
  *        to clean up its own state.
  * @return Void.
  */
-void kill_worker(LocalSchedulerClient *worker, bool cleanup) {
+void kill_worker(LocalSchedulerState *state,
+                 LocalSchedulerClient *worker,
+                 bool cleanup) {
   /* Erase the local scheduler's reference to the worker. */
-  LocalSchedulerState *state = worker->local_scheduler_state;
-  int num_workers = utarray_len(state->workers);
-  for (int i = 0; i < utarray_len(state->workers); ++i) {
-    LocalSchedulerClient *active_worker =
-        *(LocalSchedulerClient **) utarray_eltptr(state->workers, i);
-    if (active_worker == worker) {
-      utarray_erase(state->workers, i, 1);
-    }
-  }
-  /* Make sure that we erased exactly 1 worker. */
-  CHECKM(!(utarray_len(state->workers) < num_workers - 1),
-         "Found duplicate workers");
-  CHECKM(utarray_len(state->workers) != num_workers,
-         "Tried to kill worker that doesn't exist");
+  auto it = std::find(state->workers.begin(), state->workers.end(), worker);
+  CHECK(it != state->workers.end());
+  std::swap(*it, state->workers.back());
```

Comment:
For efficiency reasons? The swap should be pretty cheap and this would add an extra `if` all of the time, right?